### PR TITLE
chore(expo-dev-menu): Drop support for configuring SDK 44 and below with Prebuild.

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

Config plugins are versioned with versioned packages, meaning they shouldn't need to be SDK versioned at all. Even if they were, SDK 44 is pretty far back. On top of that, these checks don't support UNVERSIONED.

# How

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-dev-launcher/plugin/src/withDevLauncher.ts [hunk 1]
```diff
@@ -255,14 +255,10 @@ const withDevLauncherPodfile: ConfigPlugin = (config) => {
...
 const withDevLauncher = (config: ExpoConfig) => {
-  // projects using SDKs before 45 need the old regex-based integration
-  // TODO: remove these once we drop support for SDK 44
-  if (config.sdkVersion && semver.lt(config.sdkVersion, '45.0.0')) {
-    config = withDevLauncherActivity(config);
-    config = withDevLauncherApplication(config);
-    config = withDevLauncherPodfile(config);
-    config = withDevLauncherAppDelegate(config);
-  }
+  config = withDevLauncherActivity(config);
```
Comment (douglowder) [inline packages/expo-dev-launcher/plugin/src/withDevLauncher.ts:258]: I'm confused, it looks like you are taking the pre-45 behavior and making it happen always?

File: packages/expo-dev-menu/plugin/src/withDevMenu.ts [hunk 2]
```diff
@@ -119,13 +118,9 @@ const withDevMenuPodfile: ConfigPlugin = (config) => {
 };
 
 const withDevMenu = (config: ExpoConfig) => {
```
Comment (douglowder) [inline packages/expo-dev-menu/plugin/src/withDevMenu.ts:120]: Same comment as above

**Review by douglowder — COMMENTED** [review]
I want to make sure I understand -- it looks like you are taking the code path for versions < 45, and making it the default for all versions?

File: packages/expo-dev-launcher/plugin/src/withDevLauncher.ts (hunk 1 above)
Comment (EvanBacon) [reply packages/expo-dev-launcher/plugin/src/withDevLauncher.ts:258]: Good catch, looks like the plugins can be deleted.

**Review by douglowder — APPROVED** [review]

**Pull request merged**